libp2p/peerinfo.nim

```diff
@@ -49,7 +53,8 @@ func shortLog*(p: PeerInfo): auto =
 chronicles.formatIt(PeerInfo): shortLog(it)
 
 proc update*(p: PeerInfo) {.async.} =
-  p.addrs = p.listenAddrs
+  if p.addrs.len == 0 or p.addressMappers.len == 0:
```

Comment:
I'm not sure I understand this condition. Could you explain?

Reply:
Thanks for pointing this out. I should have added a comment, will do it now.

`p.addrs.len == 0` overrides `addrs` only if it is the first time `update` is being executed or if the field is empty. 
` p.addressMappers.len == 0` is for when all `addressMappers` have been removed, and we wish to have `addrs` in its initial state, i.e., a copy of `listenAddrs`.

Does it make sense?